core/tests/protractor_utils/AdminPage.js

```diff
@@ -268,6 +306,139 @@ var AdminPage = function() {
+    await miscTabButton.click();
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.uploadTopicSimilarities =
+    async function(relativePathToSimilaritiesFile) {
+      absPath = path.resolve(__dirname, relativePathToSimilaritiesFile);
+      await waitFor.visibilityOf(chooseSimilarityFileInput,
+        'Similarity upload form is not visible');
+      await chooseSimilarityFileInput.sendKeys(absPath);
+      await waitFor.elementToBeClickable(similarityFileUploadButton,
+        'Upload button taking too long to be clickable');
+      await similarityFileUploadButton.click();
+    };
+
+  this.expectSimilaritiesToBeUploaded = async function() {
+    await waitFor.textToBePresentInElement(statusMessage,
+      'Topic similarities uploaded successfully.');
+    return true;
+  };
+
+  this.expectUploadError = async function() {
+    await statusMessage.getText().then((text) => {
+      if (text.match('Server')) {
+        return true;
+      }
+      return false;
+    });
+  };
+
+  this.downloadSimilarityFile = async function() {
+    await waitFor.elementToBeClickable(similarityDownloadButton,
+      'Download similarity file button not clickable');
+    await similarityDownloadButton.click();
+  };
+
+  this.expectFileToBeDownloaded = async function() {
+    var filename = global.downloadsPath + '/topic_similarities.csv';
+    browser.driver.wait(function() {
+      return fs.existsSync(filename);
+    }, 30000);
+  };
+
+  this.clearSearchIndex = async function() {
+    await waitFor.elementToBeClickable(searchIndexClearButton,
+      'Clear search index button not clickable');
+    await searchIndexClearButton.click();
+    await browser.switchTo().alert().accept();
+  };
+
+  this.expectSearchIndexToBeCleared = async function() {
+    await waitFor.textToBePresentInElement(statusMessage,
+      'Index successfully cleared.');
+    return true;
+  };
+
+  this.flushMigrationBotContributions = async function() {
+    await waitFor.elementToBeClickable(flushMigrationBotContributionsButton,
+      'Migration bot flush contributions button not clickable');
+    await flushMigrationBotContributionsButton.click();
+    await browser.switchTo().alert().accept();
+  };
+
+  this.expectMigrationBotContributionsToBeFlushed = async function() {
+    await waitFor.textToBePresentInElement(statusMessage,
+      'Migration bot contributions successfully flushed.',
+      'Migration bot cuntributions not flushing.');
+    return true;
+  };
+
+  this.fillExtractDataForm = async function(expID, expVer, state, ans) {
+    await waitFor.pageToFullyLoad();
+    await extractDataExplorationIdInput.sendKeys(expID);
+    await extractDataExplorationVersionInput.sendKeys(expVer);
+    await extractDataStateNameInput.sendKeys(state);
+    await extractDataNumAnswersInput.sendKeys(ans);
+    await waitFor.elementToBeClickable(extractDataFormSubmitButton,
+      'Extract data form submit button not clickable');
+    await extractDataFormSubmitButton.click();
```

Comment:
Make sure to close tabs after you open them

Reply:
I've added that to the fillExtractDataForm() function.